File: backend/src/main/java/com/core/service/AuthService.java
package com.core.service;

import com.core.dto.request.LoginRequest;
import com.core.dto.request.RegisterRequest;
import com.core.dto.response.AuthResponse;

public interface AuthService {

    AuthResponse login(LoginRequest request);

    AuthResponse register(RegisterRequest request);

    AuthResponse refreshToken(String refreshTokenValue);

    void logout(String refreshTokenValue);

    /**
     * Generate a new refresh token string for a user.
     * Stores the token in the database and returns the raw token value.
     */
    String createRefreshToken(Long userId);
}

File: frontend/src/features/auth/authService.ts
import axiosInstance from '../../services/axiosInstance';
import type { LoginRequest, RegisterRequest, AuthResponse, BaseResponse } from './authTypes';

const AUTH_URL = '/auth';

export const authService = {
  login: (data: LoginRequest) =>
    axiosInstance.post<BaseResponse<AuthResponse>>(`${AUTH_URL}/login`, data),

  register: (data: RegisterRequest) =>
    axiosInstance.post<BaseResponse<AuthResponse>>(`${AUTH_URL}/register`, data),

  refreshToken: () =>
    axiosInstance.post<BaseResponse<AuthResponse>>(`${AUTH_URL}/refresh-token`),

  logout: () =>
    axiosInstance.post(`${AUTH_URL}/logout`),
};
